bundles/org.openhab.binding.shelly/README.md

```diff
@@ -0,0 +1,514 @@
+|password          |Password for HTTP authentication*                             |    no   |none                                              |
+|lowBattery        |Threshold for battery level. Set alert when level is below.   |    no   |20 (=20%), only for battery powered devices       |
+|updateInterval    |Interval for the background status check in seconds.          |    no   |1h for battery powered devices, 60s for all others|
+|eventsButton      |true: register event "trigger when a button is pushed"        |    no   |false                                             |
+|eventsPush        |true: register event "trigger on short and long push"         |    no   |false                                             |
+|eventsSwitch      |true: register event "trigger of switching the relay output"  |    no   |true                                              |
+|eventsSensorReport|true: register event "posted updated sensor data"             |    no   |true for sensor devices                           |
+|eventsCoIoT       |true: Listen for CoIoT/COAP events                            |    no   |true for battery devices, false for others        |
+
+Independent from the updateInterval setting the binding will perform a settings refresh from the device once per minute.
+There is no event from the device that the settings have be changed by the Shelly App so this is kind of a fallback to get updates at least once per minute. 
+
+
+
+## Channels
+
+### General Notes
+
+- Various channels are only visible for linking when you click Show More in the channel definition
+- input channel is only available with firmware > 1.5.6, otherwise NULL
+- use channel rollerpos only if you need the inverted value, otherwise use the control channel and Item Type Number, see example
+- Short push and long push events require firmware version 1.5.6+.
+- The different devices have different types of power meters = different set of channels.
+
+Every device has a channel group "device" with the following channels:
+
+|Group     |Channel      |Type     |read-only|Desciption                                                                       |
+|----------|-------------|---------|---------|---------------------------------------------------------------------------------|
+|device    |uptime       |Number   |yes      |Number of seconds since the device was powered up                                |
+|          |lastUpdate   |DateTime |yes      |Timestamp of last update on any of the device's channel                          |
+|          |signal       |Number   |yes      |WiFi signal strength (RSSI)                                                      |
+|          |lastAlarm    |String   |yes      |Most recent alarm for health check                                               |
+
+
+
+### Event Channel
+
+The binding maps the inbound HTTP events from the device into trigger events for those enabled in the Thing Configuration (see above).
```

Comment:
I am not really happy with having all those events unsorted in a single channel and would strongly advice to refactor this.
The buttons should clearly make use of the `DefaultSystemChannelTypeProvider.SYSTEM_BUTTON` channel type - having short and long press makes them a perfect match. For the user, this will mean that he can directly use all associated profiles for such buttons.
The relais should be a simply state channel, nothing else -> remove the events
The same for the roller shutter events - they should simply update the state of a rollershutter channel.  -> remove the events
The alarm should be a separate event channel with different events for the different alarm types.

Reply:
Those are events, not states
When you press the button for a short period of time it triggers an http event shortpush, which gets mapped to the event channel. If you release the button this event is over, it's no longer pressed, the device doesn't send an event etc. So it's an event The only thing I'm doing is passing those events to a trigger channel. Also the Harmony binding has such triggers.

Reply:
> The only thing I'm doing is passing those events to a trigger channel.

Correct, that's why I asked you to use the `DefaultSystemChannelTypeProvider.SYSTEM_BUTTON` system channel type for exactly this, which is an event channel type, not a state channel type.
But the relais + rollershutter events are state changes and thus belong to the state channels (for which state change events are readily available).